nix/flake-schemas/flake.nix

```diff
@@ -33,11 +33,59 @@
               })
             output);
       };
+      nixosConfigurationsSchema = {
```

Comment:
These three schemas already exist upstream,
https://github.com/DeterminateSystems/flake-schemas/blob/main/flake.nix

1. Why duplicate them here?
2. If this a fork, we should open PRs upstream.

Reply:
These exist because of `evalOnAllSystems = true;` config, which is supported on `juspay/inspect`

Reply:
I will add comments

Reply:
Upstream `inspect` by default recursively evaluates the derivations of all the systems supported by a given flake, which can be problematic when IFD is involved. I can’t use `builtins.tryEval` either because IFD errors are not thrown from the nix expressions language.

To avoid errors due to IFD, I introduced `systems` flake input. This means no more showing `nixosConfigurations` on darwin or vice-versa, because the flake outputs are filtered based on `forSystems` attribute in the schema of a given output, which is usually `x86_64-linux` or `aarch64-linux`. This is why `evalOnAllSystems` exists: to bypass `forSystems` based filtering on  a given schema. 

I am not pleased with this approach, let me know if this can be solved better

Reply:
> I am not pleased with this approach, let me know if this can be solved better

You could open an issue on upstream repo to initiate discussion among larger audience, as well as propose your `evalOnAllSystems` change as a PR linking to the issue.